Add recipe for geiser-stklos

jpellegrini opened the pull request:
### Brief summary of what the package does

Another of the Geiser implementations being extracted. Please see
discussion in PR #7462 (geiser-mit), and the already accepted  geiser-chibi,
geiser-chez and geiser-racket, in the same family.

### Direct link to the package repository

https://gitlab.com/emacs-geiser/stklos

### Your association with the package

I am the author and maintainer of the Geiser-STklos package, and also a contributor to [STklos Scheme](https://github.com/egallesio/STklos).

### Relevant communications with the upstream package maintainer

Not necessary

### Checklist

<!-- Please confirm with `x`: -->

- [X] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [X] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [X] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [X] My elisp byte-compiles cleanly
- [X] `M-x checkdoc` is happy with my docstrings
- [X] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

* except for `checkdoc` false positives.

riscy commented:
Thanks for this.  Since this code is already indexed, the only blocker here is with regard to a repo license.  The rest (below) is for your information --

- Add a LICENSE (or COPYING) file to the repository (ideally one GitLab can auto-detect)

But please also take a look at these --

byte-compile (using Emacs 27.1):
```
geiser-stklos.el:233:1:Warning: Unused lexical argument `module'
geiser-stklos.el:287:1:Warning: Unused lexical argument `remote'
```
These are easy to fix, by prefixing them with an underscore (`_module`), assuming they are intentionally unused.

checkdoc (using version 0.6.2):
```
Warning (emacs): 
geiser-stklos.el:110: First sentence should end with punctuation
Warning (emacs): 
geiser-stklos.el:158: Messages should start with a capital letter
```
I notice a typo on line 110; the way I deal with the one on line 158 is to use ``` `quoting' ``` around the symbol name, but that one's minor.  And some other suggestions:

- geiser-stklos.el#L123: It's safer to sharp-quote function names; use `#'`
- geiser-stklos.el#L136: It's safer to sharp-quote function names; use `#'`
- geiser-stklos.el#L156: Optionally use `unless ...` instead of `when (not ...)`
- geiser-stklos.el#L199: No `format` required; `message` takes an f-string

jpellegrini commented:
@riscy thanks a lot! I've fixed all the issues you pointed out. 
I'm sorry, I though I had fixed all docstrings before, and only one false positive remained, but I was mistaken. It seems to be ok now.

riscy commented:
Thanks!
